python_files/generate_supplementary_plots.py

```diff
@@ -26,8 +112,77 @@
 
 
 # Cell identification and classification
-
-
+def stitch_before_after_norm(
+    pre_norm_dir: Union[str, pathlib.Path], norm_dir: Union[str, pathlib.Path],
+    save_dir: Union[str, pathlib.Path],
+    run_name: str, channel: str, pre_norm_subdir: str = "", norm_subdir: str = "",
+    padding: int = 25, font_size: int = 100):
+    """Generates two stitched images: before and after normalization
+
+    pre_norm_dir (Union[str, pathlib.Path]):
+        The directory containing the run data before normalization
+    norm_dir (Union[str, pathlib.Path]):
+        The directory containing the run data after normalization
+    save_dir (Union[str, pathlib.Path]):
+        The directory to save both the pre- and post-norm tiled images
+    run_name (str):
+        The name of the run to tile, should be present in both pre_norm_dir and norm_dir
+    channel (str):
+        The name of the channel to tile inside run_name
+    pre_norm_subdir (str):
+        If applicable, the name of the subdirectory inside each FOV folder of the pre-norm data
+    norm_subdir (str):
+        If applicable, the name of the subdirectory inside each FOV folder of the norm data
+    padding (int):
+        Amount of padding to add around each channel in the stitched image
+    font_size (int):
+        The font size to use for annotations
+    """
+    # verify that the run_name specified appears in both pre and post norm folders
+    all_pre_norm_runs: List[str] = list_folders(pre_norm_dir)
+    all_norm_runs: List[str] = list_folders(norm_dir)
+    verify_in_list(
+        specified_run=run_name,
+        all_pre_norm_runs=all_pre_runs
+    )
+    verify_in_list(
+        specified_run=run_name,
+        all_post_norm_runs=all_norm_runs
+    )
+
+    # verify save_dir is valid before defining the save paths
+    validate_paths([save_dir])
```

Comment:
So we're assuming the save_dir already exists or this will fail right?

Reply:
Yes, it should exist.